### ISSUE TITLE
[Naga] ModuleInfo's entry point visibility is too restrictive

### ISSUE THREAD

**Issue by bavalpey** [gfx-rs/wgpu#6489]
**Is your feature request related to a problem? Please describe.**
I use Naga to parse wgsl files and perform some processing. However, I am disappointed with the visibility choices for much of the module. While I understand restricting the visibility of portions of Naga that are subject to change, there are a few instances where this is too much.

The instance I am focusing on here is the fact that it's not possible to get the `FunctionInfo` for an `EntryPoint` after invoking a validator. While we can get the validation information for functions, due to the support for `Index<Handle<Function>>`, there is no such trait implementation that allows us to get the function info for entry points.

**Describe the solution you'd like**
The `EntryPoint` field for `ModuleInfo` should either be marked public, or there should be a getter method for accessing the FunctionInfo.

**Describe alternatives you've considered**
N/A. The alternative would be for me to fork naga and use it.

**Additional context**
Just do this...
![image](https://github.com/user-attachments/assets/d74cd60a-e73e-4e2e-bb5f-0a17f3e332bf)

Overall, the visibility within Naga is far too restrictive and could seriously do with some additional considerations for exposing certain parts as public.
If a component is unstable, then use macros to mark this, or hide their public exposure behind a feature flag.
Right now, however, I have a naga fork just to change the visibility of its structures.

**Comment by teoxoy** [comment]
I think it would be fine to make more things public or have accessors for them if users need this data. Thoughts @jimblandy?

**Comment by jimblandy** [comment]
I think in general we do want to expose the results of validation.

**Comment by jimblandy** [comment]
Yeah, not providing this info for entry points, but providing it for functions, makes no sense.

**Comment by bavalpey** [comment]
So there actually is an interface for entry points, though I had missed it.
Specifically, there's a `get_entry_point` method defined on `ModuleInfo` that takes a usize index.

I think, though, that the point about visibility still stands.

E.g., why is `ExpressionInfo`'s `assignable_global` private?
